src/state/cached_state.rs

```diff
@@ -214,15 +188,13 @@ impl<T: StateReader> State for CachedState<T> {
         class_hash: &ClassHash,
         contract_class: &ContractClass,
     ) -> Result<(), StateError> {
-        match self.contract_classes.as_mut() {
-            Some(x) => {
-                x.insert(*class_hash, contract_class.clone());
-            }
-            None => {
-                self.contract_classes = Some(HashMap::new());
-                self.set_contract_class(class_hash, contract_class)?;
-            }
-        }
+        // TODO: Could this method's signature use `CompiledClass` instead of `ContractClass`? Or
+        //   even better, could the entire trait use `Arc<CompiledClass>`?
```

Comment:
`CompiledClass` sounds good. Since the variants are both `Arc<_>` it sounds pretty pointless to also wrap it inside an `Arc<_>`.

Reply:
Fixed. I've removed one of the two methods to set the contract class and modified the other to accept a `&CompiledClass` instead.